src/types.rs

```diff
@@ -114,25 +139,25 @@ impl LakeConfigBuilder {
 
 #[allow(clippy::enum_variant_names)]
 #[derive(thiserror::Error, Debug)]
-pub enum LakeError<E> {
+pub enum LakeError {
     #[error("Failed to parse structure from JSON: {error_message:?}")]
     ParseError {
         #[from]
         error_message: serde_json::Error,
     },
-    #[error("AWS S3 error: {error:?}")]
-    AwsError {
+    #[error("Get object error: {error:?}")]
+    S3GetError {
+        #[from]
+        error: GetObjectBytesError,
+    },
+    #[error("List objects error: {error:?}")]
+    S3ListError {
         #[from]
-        error: aws_sdk_s3::error::SdkError<E>,
+        error: ListCommonPrefixesError,
     },
     #[error("Failed to convert integer: {error:?}")]
     IntConversionError {
         #[from]
         error: std::num::TryFromIntError,
     },
-    #[error("AWS Smithy byte_stream error: {error:?}")]
-    AwsSmithyError {
```

Comment:
`S3GetError`/`S3ListError` now contain both `AwsError` and `AwsSmithyError` as discussed in "Type Erasure".